### PR TITLE
Use the same deviceId after login

### PR DESCRIPTION
**Pull request by richvdh**
There are two commits here, one which does a refactor, and another which makes the actual change. Might be easier to review them separately.

Requires https://github.com/matrix-org/matrix-js-sdk/pull/152.

### CODE REVIEW

File: src/MatrixClientPeg.js [hunk 1]
```diff
@@ -24,10 +24,6 @@ function deviceId() {
     // XXX: is Math.random()'s deterministicity a problem here?
     var id = Math.floor(Math.random()*16777215).toString(16);
     id = "W" + "000000".substring(id.length) + id;
-    if (localStorage) {
-        id = localStorage.getItem("mx_device_id") || id;
-        localStorage.setItem("mx_device_id", id);
-    }
     return id;
 }
```
Comment (dbkr) [inline src/MatrixClientPeg.js]: Maybe rename this to `generateDeviceId()` now it no longer gets 'the' device id?

File: src/components/structures/MatrixChat.js [hunk 2]
```diff
@@ -138,6 +138,19 @@ module.exports = React.createClass({
...
+        return {
+            userId: client.credentials.userId,
+            accessToken: client.getAccessToken(),
+            homeserverUrl: client.getHomeserverUrl(),
+            identityServerUrl: client.getIdentityServerUrl(),
+            guest: client.isGuest(),
+            deviceId: client.getDeviceId(),
+        }
+    },
+
```
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: Any reason for this living in MatrixChat & removing the one in MatrixClientPeg? I recently added the one in MatrixClientPeg to try & get stuff out of MatrixChat (although it wasn't yet used anywhere).

**Comment by richvdh** [comment]
This doesn't seem to do what it's supposed to. Let me have another go at it.

**Comment by richvdh** [comment]
Actually I'm going to abort it for now, pending Dave's work on making login and logout not be a disaster.

**Pull request closed without merging**